doc/explanation/package-management.md

```diff
@@ -0,0 +1,226 @@
+outputs; no state is held between the solver runs.
+
+This can lead to more packages being installed (as opam won't install new package versions
+by default if the existing versions satisfy the
+constraints), but it avoids interference from already installed packages that
+lead to potentially different solutions.
+
+After solving is done, the solution gets written into the lock directory with
+all the metadata necessary to build and install the packages. From this
+point on, there is no need to access any package repositories.
+
+:::{note}
+Solving and locking does not download the package sources. These are downloaded
+in a later step.
+:::
+
+## Building
+
+When building, Dune will read the information from the lock directory and set
+up rules to use the packages. Check {doc}`/explanation/mental-model` for
+details about rules.
+
+The rules that the package management sets up include:
+
+* Fetch rules to download the sources as well as any additional sources like
+  patches and unpack them
+* Build rules to evaluate the build instructions from the build instructions
+  stored in the lock directory
+* Install rules to put the artifacts that were built into the appropriate
+  Dune-managed folders
+
+Creating these processes as rules mean that they will only be executed on
+demand, so if the project has already downloaded the sources, it does not
+need to download them again. Likewise, if packages are installed, they stay
+installed.
+
+The results of the rules are stored in the project's `_build` directory
+and managed automatically by Dune. Thus, when cleaning the build directory, the
+installed packages are cleaned as well and will be reinstalled at the next
+build.
+
+When building the users project, the installed packages are added to the
+necessary search paths, so user code can use the dependencies without any
+additional configuration.
+
+(compatibility)=
+## Packaging for Dune Compatibility
+
+Dune can build and install most packages as dependencies, even if they are not
+built with Dune themselves. Dune will execute the build instructions from the
+lock file, very similar to opam.
+
+However, packages must meet certain requirements to be compatible with Dune.
+
+The most important one is that the packages must not use absolute paths to
+refer to files. That means, they cannot read the path they are being built or
+installed into and expect this path to be stable. Dune builds packages in a
+sandbox location, and after the build has finished, it moves the files to the
+actual destination.
+
+The reason for this is clear. On one hand it enables building without messing up the
+current state, and on the other hand it allows for caching artifacts across
+projects.
+
+To sidestep these restructions in many cases the solution is to use relative
+paths, as Dune guarantees that packages installed into different sections are
+installed in a way where their relative location stays the same.
```

Comment:
```suggestion
To work around these restrictions, a common solution is to use relative paths. Dune ensures that packages installed in different sections maintain the same relative location, allowing for more flexibility.
```
Is this still accurate?

Reply:
It wouldn't say "side step" or "work around" to refer to the expected way of satisfying a condition. Maybe:

To comply with this requirement, the usual solution ...

Reply:
I like the "comply" wording because it makes it sound positive, whereas "work around" and "side step" is a sounds like there is a bug that people have to try to avoid. I would argue that avoiding absolute paths is a good thing in general and makes the code better even without Dune package management.